components/scream/src/share/atm_process/atmosphere_process_group.hpp

```diff
@@ -65,13 +65,17 @@ class AtmosphereProcessGroup : public AtmosphereProcess
   void set_required_field (const Field<const Real>& field);
   void set_required_group (const FieldGroup<const Real>& group);
 
+  // Gather internal fields from all processes in the group
+  void gather_internal_fields ();
```

Comment:
Any rules about when this can/can't be called?

Reply:
Ah, yes, this must be called before any attempt to query the list of internal fields for the group. I will add a note in the comment.